lib/validation/validateNode.js

```diff
@@ -103,27 +103,32 @@ function onlyLine(opts: Options) {
         validate(node) {
             const { nodes } = node;
 
-            const toWrap = [];
-            const toRemove = [];
+            let toAdd = List();
+            const toRemove = List();
 
             nodes.forEach(child => {
-                if (child.kind === 'text') toWrap.push(child);
-                else if (child.type !== opts.lineType) toRemove.push(child);
+                if (child.kind === 'text') {
+                    const lines = deserializeCode(opts, child.text).nodes;
+                    toAdd = toAdd.concat(lines);
```

Comment:
That's great. I'm sorry the plugin was not doing this already.
I think we need to save here the index of the created lines in the `code_block` as well. This is to support the case of a mix of `text`, `code_line` and `invalid_block` inside a `code_block`. Otherwise the deserialized lines could end up at the wrong index. Could you add a test for this case ?

Reply:
I agree this change needs to be made.

I refactored to allow for it but my new mixed test is failing in a strange way. When I log inside `validateNode` it seems as though the input data is only coming through as a single node, but I can't see what I've done wrong. If you have the time i'd love a second pair of 👀